src/value_generators/src/DocumentGenerator.cpp

```diff
@@ -736,6 +739,33 @@ class RandomDateGenerator : public Generator<bsoncxx::types::b_date> {
     const std::chrono::milliseconds _max;
 };
 
+
+class IncGenerator : public Generator<int64_t> {
+public:
+    IncGenerator(const Node& node, GeneratorArgs generatorArgs)
+        : _actorId{generatorArgs.actorId},
+          _counter{generatorArgs.counter},
+          _step{extract(node, "step", "^Inc").to<int64_t>()},
+          _multiplier{extract(node, "multiplier", "^Inc").to<int64_t>()}
+          {}
+
+    int64_t evaluate() override {
+//        std::lock_guard<std::mutex> lg(mtx);
+        std::cout << "Counter before " << _counter << std::endl;
+        _counter=_actorId*_multiplier + _counter + _step;
```

Comment:
The _actorId*multiplier should be constant. You could make that a member of the generator.